### PR TITLE
Implementation of UNET on ISIC data set for generating lesion segmentation and comparing with actual in terms of Dice Score

### PR DESCRIPTION
**Pull request by Dattatreya-45586327**
**Name - Dattatreya Majumdar
Student ID - 45586327**

- This project is dedicated to implement **UNET** on the ISIC data set to generate predicted segmentations on lesion images and comparing the predicted images with the actual segmentations using the DICE score and achieving minimum Dice Score of 0.7 on the overall test data set. The train, validation and test data sets are split in 60:20:20 ratio. The dice loss and accuracy plots for training and validation data sets wrt epochs is provided.
- The project comprises of 5 python files with the Driver_Script.py implementing all functionalities using command line interface
- The README.md file summarizes the overall objective and functionality of the individual files.
- Apart from that there are two other sample output files - A CSV file enlisting the individual dice coefficient of the test data a set and a PNG file showing the actual lesion image, corresponding actual segmentation image and predicted segmentation image by the UNET model

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
- Reasonably detained commit messages
- Reported min DSC of 0.7, avg seem much higher
- Code and results appear valid for Easy Difficulty

Other Feedback
- Group other files other than code and put them in sub folders (e.g. `images\*.png`)
- Incorrect python file naming convention, follow e.g. `lorem_ipsum.py`
- clean up unused lines of code and imports e.g. `from keras.models import Model, load_model
from keras.layers import Input, BatchNormalization, Activation, Dense, Dropout
from keras.layers.core import Lambda, RepeatVector, Reshape
from keras.layers.convolutional import Conv2D, Conv2DTranspose
from keras.layers.pooling import MaxPooling2D, GlobalMaxPool2D
from keras.layers.merge import concatenate, add
from keras.callbacks import EarlyStopping, ModelCheckpoint, ReduceLROnPlateau
from keras.optimizers import Adam
from keras.preprocessing.image import ImageDataGenerator, array_to_img, img_to_array, load_img
from keras.utils import to_categorical
from keras import backend as K
`

**Comment by rexsw** [comment]
Code looks good for easy difficulty

Comments such as:
     #Function for evaluating dice coefficient for the overall ISIC test data set
     def dice_coeff(y_true, y_pred, smooth=1):
Don’t add much to the readability of the code and such be left out. And comments such as:
    # Function for extracting the ISIC test image file names
    def img_test_fn(seg_path):
Could be left out if the function name was improved.

Using a whole large library like pandas to do one thing isn’t the best idea and makes installing and maintaining your code harder. Sometimes it’s better to just take care of it yourself. 

Tensorflow has lots of image processing functions built into it, which you likely could have used instead of scikitimage, again the fewer libraries you depend on the easier your project is to maintain.

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme OK
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used 
## Documentation
Description and Comments good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
**Feedback required. remove unnecessary imports.**
In the future, a more detailed pull request could've been done. Remember you could use/reuse the README info.

**Comment by Dattatreya-45586327** [comment]
Name - Dattatreya Majumdar
Student ID - 45586327
I have removed the unnecessary package imports from each of the .py file and have created a separate folder for the images. The corresponding changes have also been reflected in the paths of the images in the README.md file.
